liblouis/compileTranslationTable.c

```diff
@@ -4539,18 +4539,23 @@ resolveSubtable(const char *table, const char *base, const char *searchPath) {
 	char *tableFile;
 	static struct stat info;
 
-	if (table == NULL || table[0] == '\0') return NULL;
+	if (table == NULL || table[0] == '\0' ||
+			strlen(table) >= MAXSTRING * sizeof(char) * 2)
```

Comment:
Can you put `MAXSTRING * sizeof(char) * 2` into a define of a constant?

Reply:
I define a MAX_TABLEFILE_SIZE in resolveSubtable().

Reply:
Do you think the expression needs to be in brackets? I'm not totally clear on the precedence rules but if you have something like `if (foo >= MY_CONST) ...` and `MY_CONST` is expanded from a `#define` would it not be more prudent to define `MY_CONST` using brackets, e.g. `#define MAX_TABLEFILE_SIZE (MAXSTRING * sizeof(char) * 2)`

This is not a trick question. I genuinely do not know.

Ah, I just found these [docs on Operator Precedence Problems](https://gcc.gnu.org/onlinedocs/cpp/Operator-Precedence-Problems.html#Operator-Precedence-Problems). So maybe parentheses around the entire macro definition would prevent any problems.

Reply:
Yes, you are right!
Parentheses could prevent unexpected behaviors to happen, I should have noticed it.
Now they are added for the define expression.

Reply:
Excellent thanks